pkg/registry/generic/etcd/etcd.go

```diff
@@ -159,12 +159,20 @@ func (e *Etcd) NewList() runtime.Object {
 }
 
 // List returns a list of items matching labels and field
-func (e *Etcd) List(ctx api.Context, label labels.Selector, field fields.Selector, options *api.ListOptions) (runtime.Object, error) {
-	return e.ListPredicate(ctx, e.PredicateFunc(label, field), options)
+func (e *Etcd) List(ctx api.Context, options *api.ListOptions) (runtime.Object, error) {
+	label := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		label = options.LabelSelector
+	}
+	field := fields.Everything()
+	if options != nil && options.FieldSelector != nil {
+		field = options.FieldSelector
+	}
+	return e.listPredicate(ctx, e.PredicateFunc(label, field), options)
 }
 
-// ListPredicate returns a list of all the items matching m.
-func (e *Etcd) ListPredicate(ctx api.Context, m generic.Matcher, options *api.ListOptions) (runtime.Object, error) {
+// listPredicate returns a list of all the items matching m.
+func (e *Etcd) listPredicate(ctx api.Context, m generic.Matcher, options *api.ListOptions) (runtime.Object, error) {
```

Comment:
We use ListPredicate in various contexts - it needs to remain public.  We can make it private in a separate followup.